core/src/vaults.ts

```diff
@@ -0,0 +1,335 @@
+import getContract, { getContractInterfaceByName } from './contracts';
+import getProvider from './provider';
+import {
+    VaultBase,
+    CollateralType,
+    VaultAmount,
+    VaultCollateralParameters,
+    Vault,
+    VaultTransactionNotLiquidated,
+    VaultTransaction,
+} from './types';
+import BigNumber from './bignumber';
+import { ethers } from 'ethers';
+import {
+    DAI_NUMBER_OF_DIGITS,
+    RAD_NUMBER_OF_DIGITS,
+    RAY_NUMBER_OF_DIGITS,
+    WAD_NUMBER_OF_DIGITS,
+} from './constants/UNITS';
+import { getApproximateLiquidationFees } from './fees';
+import { fetchDateByBlockNumber } from './date';
+import memoizee from 'memoizee';
+import getSigner from './signer';
+import executeTransaction from './execute';
+import { getOsmPrices } from './oracles';
+
+const CACHE_EXPIRY_MS = 60 * 1000;
+
+const _fetchVaultBase = async (network: string, id: number): Promise<VaultBase> => {
+    const contract = await getContract(network, 'CDP_MANAGER');
+    const address = await contract.urns(id);
+    if (address === '0x0000000000000000000000000000000000000000') {
+        throw new Error('Vault does not exist');
+    }
+    const collateralTypeHex = await contract.ilks(id);
+    const collateralType = ethers.utils.parseBytes32String(collateralTypeHex);
+    return {
+        id,
+        address,
+        collateralType,
+        lastSyncedAt: new Date(),
+        network,
+    };
+};
+
+export const fetchVaultBase = memoizee(_fetchVaultBase, {
+    maxAge: CACHE_EXPIRY_MS,
```

Comment:
As far as I understand, those values would never change (except for the `lastSyncedAt`), right? So maybe you can remove maxAge here, but move `lastSyncedAt` to the parent?

Reply:
This comment was not addressed

Reply:
> parent

to the child you mean? the vault type that inherits from the base which has the data updated regularly.

Reply:
No, to the parent! Currently, the `lastSyncedAt` will be cached forever and will be shown on the frontend as `last fetched at`, which will be misleading, since we refetch all other things about the vault. The proposal is to change the VaultBase type to exclude `lastSyncedAt` since it's the only value that is changeable in this function and move this to the parent logic (enrichment?)

We can leave this to tackle in the next PR, since it's a minor thing